pkg/virt-controller/watch/vmi.go

```diff
@@ -1305,6 +1293,28 @@ func (c *VMIController) virtlauncherAttachmentPods(virtlauncherPod *k8sv1.Pod) (
 	return attachmentPods, nil
 }
 
+func (c *VMIController) waitForFirstConsumerTemporaryPods(vmi *virtv1.VirtualMachineInstance) ([]*k8sv1.Pod, error) {
+	var temporaryPods []*k8sv1.Pod
+
+	// Get all pods from the namespace
+	pods, err := c.listPodsFromNamespace(vmi.Namespace)
+	if err != nil {
+		return temporaryPods, err
+	}
+
+	for _, pod := range pods {
+		if !controller.IsControlledBy(pod, vmi) {
```

Comment:
For me that is more readable, What do you think?
if controller.IsControlledBy(pod, vmi) && controller.IsTempPod(pod) {
  temporaryPods = append(temporaryPods, pod)
}

Reply:
but I am not sure it is a good solution

Reply:
Had to rewrite according to https://github.com/kubevirt/kubevirt/pull/5891#issuecomment-865998195
let me know what you think